### PR TITLE
ENM - Shell We Dance

### PR DESCRIPTION
**Pull request by m241dan**
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

The only thing missing is that the whole family of these mobs does not go into their shell like they are supposed to. It makes the BC a little easier than it should be but what can you do? Someone should make that mixin.

### CODE REVIEW

**Comment by m241dan** [comment]
Rebased with Release and fixed the spell_list conflict created by adding the Puddings since submitting.

File: scripts/zones/Boneyard_Gully/IDs.lua [hunk 1]
```diff
@@ -17,11 +17,32 @@ zones[tpz.zone.BONEYARD_GULLY] =
...
     mob =
-    {
+	{
     },
     npc =
     {
     },
+	shellWeDance = {
+		[1] = {
+			PARATA_ID        = 16810024,
```
Comment (TeoTwawki) [inline scripts/zones/Boneyard_Gully/IDs.lua]: them sneaky tabs gotchu

File: sql/mob_spell_lists.sql [hunk 2]
```diff
@@ -3890,7 +3890,16 @@ INSERT INTO `mob_spell_lists` VALUES ('Ebony_Pudding',430,258,7,255);  -- Bind
 INSERT INTO `mob_spell_lists` VALUES ('Ebony_Pudding',430,273,31,255); -- Sleepga
 INSERT INTO `mob_spell_lists` VALUES ('Ebony_Pudding',430,274,56,255); -- Sleepga II
 
-
+INSERT INTO `mob_spell_lists` VALUES ('Bladmall',431,46,63,255);                  -- Protect IV
+INSERT INTO `mob_spell_lists` VALUES ('Bladmall',431,133,68,255);          		 -- Shellra IV
```
Comment (TeoTwawki) [inline sql/mob_spell_lists.sql]: whitespace appears to have mixed tabs and spaces in same line here.

File: sql/mob_pool_mods.sql [hunk 3]
```diff
@@ -59,6 +59,7 @@ INSERT INTO `mob_pool_mods` VALUES (276,12,1,1);
 INSERT INTO `mob_pool_mods` VALUES (289,4,4,1);
 INSERT INTO `mob_pool_mods` VALUES (370,28,5,1);
 INSERT INTO `mob_pool_mods` VALUES (410,236,20,0);
+INSERT INTO `mob_pool_mods` VALUES (444,23,23,1); -- Bladmall immune to Sleep, Bind, Silence and Gravity.
```
Comment (m241dan) [inline sql/mob_pool_mods.sql:62]: Is this the correct way to do this? I think this needs to be changed.
@TeoTwawki

Comment (TeoTwawki) [reply sql/mob_pool_mods.sql:62]: true immunity is a field in mob pools, don't add as a mod.

high resistance will be a mod but only put here if the mob is non-scripted and it affects all mobs using the pool (if its on mob, prolly belongs in the spawn mods if non-scripted)

you can tell apart resist from immunity by the log messages in retail. 

Regular old high resistance:
_"The \<player\> casts \<spell\>. \<target\> resists the spell."_

Immunity:
_"completely resists the spell"_

Job trait proc:
_"Resist!"_

Immunobreak (for completeness):
_"resists the spell"_
_"Immunobreak!"_

**Comment by 59blargedy** [comment]
not sure if the mixin should be implemented for this script now? they may have different properties, but figured i'd ask

**Comment by zach2good** [comment]
There is a very handy mixin you can copy/steal/appropriate that handles mobs going into a shell:
https://github.com/project-topaz/topaz/pull/989

EDIT: On closer inspection, this fight is Uragnites, the mixin is Uragnites... sounds like a match to me!

**Comment by zach2good** [comment]
Merged into `feature/enm` and shifted around all the SQL required. Gonna open an issue for adding the Urganite mixin to the mobs. Thanks for the ENM!

**Pull request merged**